### DISCUSSION TITLE
How can protect API graphql

### DISCUSSION THREAD

**Discussion by IRediTOTO** [wp-graphql/wp-graphql#1492] [Q&A] [answered]
Hi, is there any feature to protect graphql api? We don't want someone crawl all our data easy, right? 
I want if we call /graphql , it will ask for username/password to get data.
I know about this plugin https://github.com/wp-graphql/wp-graphql-jwt-authentication
But it use for different purpose.

**Comment by jasonbahl** [comment, marked as answer]
There's a blog post here that proposes one option for this: https://www.wpgraphql.com/2019/01/30/preventing-unauthenticated-requests-to-your-wpgraphql-api/

You could also implement .htaccess (or similar) auth requiring a username/password for the endpoint itself, before WordPress execution even begins. 

Something like suggested here: https://stackoverflow.com/questions/5229656/password-protecting-a-directory-and-all-of-its-subfolders-using-htaccess
